primitives/merkle-mountain-range/src/lib.rs

```diff
@@ -399,6 +399,17 @@ sp_api::decl_runtime_apis! {
 		/// Generate MMR proof for a leaf under given index.
 		fn generate_proof(leaf_index: LeafIndex) -> Result<(EncodableOpaqueLeaf, Proof<Hash>), Error>;
 
+		/// Generate a historical MMR proof for a leaf with given index.
+		///
+		/// This function may be used to generate proofs using MMR state at some earlier
+		/// point. To do that, you shall provide historical value of the `leaves_count`.
+		/// The proof generated by this function may be verified by clients that only have
+		/// some old MMR root (corresponding to the `leaves_count`).
+		fn generate_historical_proof(
+			leaf_index: LeafIndex,
+			leaves_count: LeafIndex,
+		) -> Result<(EncodableOpaqueLeaf, Proof<Hash>), Error>;
```

Comment:
Would it make sense to directly add this with support for multiple leaves at once? (see https://github.com/paritytech/substrate/pull/10635)

Would the light client benefit for generating historical proofs in a batch?

Reply:
Yeah - I think it makes sense to add support for proving several leaves, even though we'll only be using this API to prove single leaf now.

Reply:
Then let's probably defer this PR until #10635 is merged to reuse primitives from there && etc